source/common/common/hash.cc

```diff
@@ -1,9 +1,31 @@
 #include "common/common/hash.h"
 
+#include "envoy/common/exception.h"
+
 #include "absl/strings/string_view.h"
 
 namespace Envoy {
 
+uint64_t HashUtil::xxHash64(const std::vector<std::string>& input, uint64_t seed) {
+  XXH64_state_t* const state = XXH64_createState();
+  if (state == nullptr) {
+    throw EnvoyException("create XXH64 state failed");
+  };
+  if (XXH64_reset(state, seed) == XXH_ERROR) {
+    XXH64_freeState(state);
+    throw EnvoyException("XH64 reset seed failed");
+  }
+  for (const auto& i : input) {
+    if (XXH64_update(state, i.c_str(), i.size()) == XXH_ERROR) {
+      XXH64_freeState(state);
+      throw EnvoyException("XH64 hash update failed");
+    };
+  }
+  auto hash = XXH64_digest(state);
+  XXH64_freeState(state);
+  return hash;
+}
```

Comment:
You can replace all of this with:
```
  absl::Hash<const std::vector<absl::string_view>> hasher;
  return hasher(v);
```
The above makes me queezy as we are calling this on the data plane and we do not allow exceptions in the data plane. See https://github.com/envoyproxy/envoy/issues/14320 and https://github.com/envoyproxy/envoy/pull/15251 

It also allocates memory and that should not be necessary to compute a hash.

Reply:
got it ,thanks